dbms/src/Storages/Transaction/TiDB.cpp

```diff
@@ -1103,4 +1104,62 @@ ColumnInfo fieldTypeToColumnInfo(const tipb::FieldType & field_type)
     return ret;
 }
 
+DAGSchema genSchemaFromTableScan(const tipb::TableScan & table_scan)
+{
+    DAGSchema schema;
+    schema.reserve(table_scan.columns_size());
+    for (Int32 i = 0; i < table_scan.columns_size(); ++i)
+    {
+        String name = "mock_table_scan_" + std::to_string(i);
+        TiDB::ColumnInfo column_info;
+        auto const & ci = table_scan.columns(i);
+        column_info.tp = static_cast<TiDB::TP>(ci.tp());
+        column_info.id = ci.column_id();
+        column_info.name = name;
+        schema.push_back({name, column_info});
+    }
+    return schema;
+}
+
+NamesAndTypes genNamesAndTypesFromTableScan(const tipb::TableScan & table_scan)
+{
+    NamesAndTypes names_and_types;
+    names_and_types.reserve(table_scan.columns_size());
+
+    for (Int32 i = 0; i < table_scan.columns_size(); ++i)
+    {
+        String name = "mock_table_scan_" + std::to_string(i);
+        TiDB::ColumnInfo column_info;
+        auto const & ci = table_scan.columns(i);
```

Comment:
use `TiDBTableScan.getColumns()` directly?

Reply:
ok, maybe I will add a new overload of the function.